src/main/java/com/amazon/ion/bytecode/bin11/bytearray/ListOpcodeHandlers.kt

```diff
@@ -0,0 +1,197 @@
+                    macroIndices,
+                    symbolTable,
+                )
+            }
+        }
+        return containerLength + prefixLength
+    }
+}
+
+/**
+ * Writes a delimited list to the bytecode buffer. Handles opcode `0xF0`.
+ */
+internal object DelimitedListOpcodeHandler : OpcodeToBytecodeHandler {
+    @OptIn(ExperimentalStdlibApi::class)
+    override fun convertOpcodeToBytecode(
+        opcode: Int,
+        source: ByteArray,
+        position: Int,
+        destination: BytecodeBuffer,
+        constantPool: AppendableConstantPoolView,
+        macroSrc: IntArray,
+        macroIndices: IntArray,
+        symbolTable: Array<String?>
+    ): Int {
+        assert(opcode == 0xf0) { "Handler cannot compile opcode ${opcode.toHexString()}" }
+
+        var p = position
+        BytecodeEmitter.emitList(destination) {
+            while (true) {
+                val opcode = source[p++].unsignedToInt()
+                if (opcode == OpCode.DELIMITED_CONTAINER_END) {
+                    break
+                }
+                p += OpcodeHandlerTable.handler(opcode).convertOpcodeToBytecode(
+                    opcode,
+                    source,
+                    p,
+                    destination,
+                    constantPool,
+                    macroSrc,
+                    macroIndices,
+                    symbolTable,
+                )
+            }
+        }
+        val bytesRead = p - position
+        return bytesRead
+    }
+}
+
+/**
+ * Writes a tagless-element list to the bytecode buffer. Handles opcode `0x5B`.
+ */
+internal object TaglessElementListOpcodeHandler : OpcodeToBytecodeHandler {
+    @OptIn(ExperimentalStdlibApi::class)
+    override fun convertOpcodeToBytecode(
+        opcode: Int,
+        source: ByteArray,
+        position: Int,
+        destination: BytecodeBuffer,
+        constantPool: AppendableConstantPoolView,
+        macroSrc: IntArray,
+        macroIndices: IntArray,
+        symbolTable: Array<String?>
+    ): Int {
+        assert(opcode == 0x5b) { "Handler cannot compile opcode ${opcode.toHexString()}" }
+
+        var p = position
+        val childOpcode = source[p++].unsignedToInt()
+        val macroAddress = when (childOpcode) {
+            in 0x00..0x47 -> childOpcode
+            in 0x48..0x4f, 0xf4 -> {
+                val addressValueAndLength = PrimitiveDecoder.readFlexUIntValueAndLength(source, p)
+                val addressValue = addressValueAndLength.toInt()
+                val addressLength = addressValueAndLength.shr(Int.SIZE_BITS).toInt()
+                p += addressLength
+                addressValue
+            }
+            else -> -1
+        }
+
+        val containerSizeValueAndLength = PrimitiveDecoder.readFlexUIntValueAndLength(source, p)
+        val containerLength = containerSizeValueAndLength.toInt()
+        val prefixLength = containerSizeValueAndLength.shr(Int.SIZE_BITS).toInt()
+        p += prefixLength
```

Comment:
Minor naming suggestions to aid with clarity.
```suggestion
        val childCountValueAndLength = PrimitiveDecoder.readFlexUIntValueAndLength(source, p)
        val childCount = childCountValueAndLength.toInt()
        val prefixSize = childCountValueAndLength.shr(Int.SIZE_BITS).toInt()
        p += prefixSize
```